ansible/roles/pgbackrest/tasks/main.yml

```diff
@@ -0,0 +1,69 @@
+- include_vars: ../../postgresql/defaults/main.yml
+
+- name: Install pgBackRest and requirements
+  apt: name={{item}} state=installed
+  with_items:
+     - pgbackrest
+
+# FIXME: only if no other standby config is in place
+#- name: Editing postgresql.conf file
+#  become: True
+#  blockinfile:
+#    dest: "{{ postgresql_config_home }}/postgresql.conf"
+#    block: |
+#      archive_command = 'pgbackrest --stanza=backup archive-push %p'
+#      archive_mode = on
+#      max_wal_senders = 3
+#      wal_level = hot_standby
+#  notify: Restart postgres
+#  when: not is_pg_standby
+
+- name: Editing archive_command postgresql.conf file
+  become: True
+  lineinfile:
+    dest: "{{ postgresql_config_home }}/postgresql.conf"
+    regexp: '^archive_command ='
+    line: "archive_command = 'rsync -a %p postgres@192.168.33.17:/opt/data/postgresql/wal_archive/%f; pgbackrest --stanza=backup archive-push %p'"
+  notify: Restart postgres
+  when: not is_pg_standby
+  
+- name: Creating pgbackrest.conf file
+  become: True
+  template:
+    src=pgbackrest.conf.j2
+    dest=/etc/pgbackrest.conf
+    owner=postgres
+    group=postgres
+    mode=640
+
+- name: Creating postgres stanza
+  become: True
+  become_user: postgres
+  vars:
+    ansible_ssh_pipelining: True
+  shell: pgbackrest --stanza=backup --log-level-console=info stanza-create
```

Comment:
Is this OK to run multiple times?

Reply:
it is, however it is marked as "changed" everytime,  i am going to check if {{postgresql_backup_dir}}/archive/backup exists before running it again.

Reply:
done. however i still have the problem of postgresql restarting twice because of the changes done to the archive_command line by the two different playbooks (postgresql and pgbackrest)

Reply:
yea, that's unfortunate